sdk/core/core/inc/az_http_policy.h

```diff
@@ -27,6 +27,17 @@
 
 typedef struct az_http_policy az_http_policy;
 
+/**
+ * @brief options for retry policy
+ * max_retry = maximun number of retry intents before returning error
+ * delay = waiting time before retrying in miliseconds
+ *
+ */
+typedef struct {
+  size_t max_retry;
```

Comment:
1. delay needs a better name as it probably represents milliseconds; so perhaps delay_in_ms
2. I think size_t bit width varies and I think it better if we just go win 16-bit integer for max_retries and 32-bit int for delay_in_ms

Reply:
yeah, I added doxygen about delay represents milliseconds. but yes, I can update to make it even easier to get it

Reply:
Yes. Completely agree. Every time we introduce physical units (like time, distance, etc.) the name should reflects the size of units. The only exception is if we use SI (seconds, kg, etc.). In this specific case, `delay` is not in seconds so it should have different name.